### PR TITLE
BL-3331 Re-enable draggable and resizable for reader setup

### PR DESCRIPTION
**Pull request by JohnThomson**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/1040)

### CODE REVIEW

File: src/BloomBrowserUI/bookEdit/toolbox/decodableReader/decodableReader.ts [hunk 1]
```diff
@@ -150,7 +156,7 @@ export function showSetupDialog(showWhat) {
                 Cancel: {
                     text: theOneLocalizationManager.getText('Common.Cancel', 'Cancel'),
                     click: function () {
-                        $(this).dialog("close");
+                      setupDialogElement.dialog("close");
```
Comment (hatton) [inline src/BloomBrowserUI/bookEdit/toolbox/decodableReader/decodableReader.ts]: I'm curious why the $(this) didn't work.

**Comment by JohnThomson** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomBrowserUI/bookEdit/toolbox/decodableReader/decodableReader.ts, line 159 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1040#-KGmmw6ENgCZjmy7JTXW-r1-159:-KGoVDPn62mzk5p7_YoS:-161593420) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c9643a17d747d5c1bb869a841a0cdd71abdfd4ee/src/BloomBrowserUI/bookEdit/toolbox/decodableReader/decodableReader.ts#L159)):*
Closest I can get is, 'this' is an element which we added to the top-level frame. This code is back in the toolbox context. $(this) creates a wrapper using the toolbox JQuery. We're trying to close a dialog created with the parent window JQuery. Have to dig into the implementatin of JQueryUI to figure out why some things work and others don't. (Should I make that a comment? I hinted at it in the comment above, but apparently not explicitly enough.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1040)*

**Comment by hatton** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomBrowserUI/bookEdit/toolbox/decodableReader/decodableReader.ts, line 159 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1040#-KGmmw6ENgCZjmy7JTXW-r1-159:-KGqh4fBW9vO5nYHpMia:126657596) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c9643a17d747d5c1bb869a841a0cdd71abdfd4ee/src/BloomBrowserUI/bookEdit/toolbox/decodableReader/decodableReader.ts#L159)):*
I'm not asking for a comment here, I don't think comments should be there to teach me JS. Rather, I'm just trying to understand something.  I would have expected that "this" would equal setupDialogElement.  In my browser, they appear to be the same:



---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1040)*

**Comment by JohnThomson** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomBrowserUI/bookEdit/toolbox/decodableReader/decodableReader.ts, line 159 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1040#-KGmmw6ENgCZjmy7JTXW-r1-159:-KGsJXCZe-6BDfZo_t9U:1585458432) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c9643a17d747d5c1bb869a841a0cdd71abdfd4ee/src/BloomBrowserUI/bookEdit/toolbox/decodableReader/decodableReader.ts#L159)):*
setupDialogElement is not this, but $(this) (a JQuery wrapper around the dialog element)...except that the $ is the parent window's static JQuery, rather than the toolbox's, which is what we get using $(this) here. The element was wrapped by the function in the parent window that created the element, rather than by a handler function that belongs to the toolbox. And that's what makes the difference, somehow.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1040)*

**Comment by hatton** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomBrowserUI/bookEdit/toolbox/decodableReader/decodableReader.ts, line 159 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1040#-KGmmw6ENgCZjmy7JTXW-r1-159:-KGsN0y6D-gbeGzA1Nh8:1759033895) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c9643a17d747d5c1bb869a841a0cdd71abdfd4ee/src/BloomBrowserUI/bookEdit/toolbox/decodableReader/decodableReader.ts#L159)):*
OK, thanks. I've added this comment to this line, in the PR I'm working on:

//nb: the element pointed to here by setupDialogElement is the same as "this"
                        //however, the jquery that you'd get by saying $(this) is *not* the same one as
                        //that stored in setupDialogElement. Ref BL-3331.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1040)*

**Pull request merged**